File: week_1/list_dir.cpp
#include <iostream>
#include <filesystem>

int main() {
    try {
        for (const auto& entry : std::filesystem::directory_iterator(".")) {
            std::cout << entry.path().filename().string() << '\n';
        }
    } catch (const std::exception& e) {
        std::cerr << "Error listing directory: " << e.what() << '\n';
        return 1;
    }
    return 0;
}

File: week_1/list_dir.rs
use std::fs;
use std::error::Error;

fn main() {
    if let Err(e) = run() {
        eprintln!("Error listing directory: {e}");
        std::process::exit(1);
    }
}

fn run() -> Result<(), Box<dyn Error>> {
    for entry in fs::read_dir(".")? {
        let entry = entry?;
        println!("{}", entry.file_name().to_string_lossy());
    }
    Ok(())
}
